pyproject.toml

```diff
@@ -20,15 +20,15 @@ use_parentheses = true
 check_untyped_defs = true
 disallow_any_generics = true
 disallow_incomplete_defs = true
-disallow_subclassing_any = true
+disallow_subclassing_any = false
```

Comment:
The `true` to `false` is from the template default. web3.py has never had an entry for that rule, and `mypy` defaults to `false`, so the errors appeared for the first time when merging the template. I figured to leave it so at least we know we are explicitly ignoring them. It's not too bad, though - only 8 errors. I could put `ignore` lines in for them and then turn it back on so we don't ignore any more in the future.

Reply:
I'm good leaving it as `false` then

Reply:
Meh, I tried it. Several errors were resolved by adding pydantic types to mypy checking. Several more will be resolved when `eth-abi` gets fully typed and we'll get an `unused ignore` warning. Seems good to have it on to avoid adding more.